src/DataStructures/DataBox/Deferred.hpp

```diff
@@ -184,10 +184,70 @@ class deferred_assoc_state : public assoc_state<Rt> {
     func_(make_not_null(&t_), retrieve_from_deferred(std::get<Is>(args_))...);
   }
 };
+
+// Specialization to handle functions that return a `const T&`. We treat the
+// return value as pointer to the data we actually want to have visible to us
+// when we retrieve the data. The `get` function dereferences the pointer we
+// store so that we have a reference to work with when retrieving the stored
+// data reference. Dereferencing the pointer ensures that all functions that use
```

Comment:
I am finding the previous two sentences (from "We treat the return value...") difficult to understand. Is this rephrasing still accurate?

"We store a pointer to the returned data, i.e., a pointer to the data we actually want to retrieve from the DataBox. The `get` function dereferences the stored pointer, recovering a reference to the data to retrieve."